source/base/parameter_handler.cc

```diff
@@ -1862,6 +1862,7 @@ namespace
 
 bool ParameterHandler::read_input_from_xml (std::istream &in)
 {
+  AssertThrow(in, ExcIO());
```

Comment:
I don't think so (technically). The `PathSearch` version (line 1700) will catch a `PathSearch`-specific exception, but also has an assertion around the file stream: if the file exists then an exception should be thrown before that assertion, so I don't think that `AssertThrow(file_stream, ExcIO())` will ever throw. I believe that this means, in practice, that if the file does not exist then no exception leaves the function.